src/vivarium_nih_us_cvd/components/treatment.py

```diff
@@ -0,0 +1,361 @@
+            != data_values.MEDICATION_ADHERENCE_TYPE.ADHERENT
+        ].index
+        ldlc_non_adherent = pop[
+            pop[self.ldlc_medication_adherence_type_column]
+            != data_values.MEDICATION_ADHERENCE_TYPE.ADHERENT
+        ].index
+        pop.loc[medicated_sbp.intersection(sbp_non_adherent), self.sbp_medication_column] = 1
+        pop.loc[
+            medicated_ldlc.intersection(ldlc_non_adherent), self.ldlc_medication_column
+        ] = 1
+
+        return pop
+
+    def calculate_medication_coverage_probabilities(self, pop: pd.DataFrame) -> pd.DataFrame:
+        """Determine the probability of each simulant being medicated"""
+
+        # Calculate the covariates
+        medication_coverage_covariates = {}
+        for (
+            name,
+            c_int,
+            c_sbp,
+            c_ldlc,
+            c_age,
+            c_sex,
+        ) in data_values.MEDICATION_COVERAGE_COEFFICIENTS:
+            medication_coverage_covariates[name] = np.exp(
+                c_int
+                + c_sbp * self.sbp(pop.index)
+                + c_ldlc * self.ldlc(pop.index)
+                + c_age * pop["age"]
+                + c_sex * pop["sex"].map(data_values.BASELINE_MEDICATION_COVERAGE_SEX_MAPPING)
+            )
+        # Calculate probabilities of being medicated
+        p_medication = {}
+        p_denominator = sum(medication_coverage_covariates.values()) + 1
+        for med_type, cov in medication_coverage_covariates.items():
+            p_medication[med_type] = pd.Series(cov / p_denominator, name=med_type)
+        p_medication["none"] = pd.Series(1 / p_denominator, name="none")
+
+        return pd.concat(p_medication, axis=1)
+
+    def on_time_step_cleanup(self, event: Event) -> None:
+        """Update treatments"""
+        pop = self.population_view.subview(
+            [
+                self.visit_type_column,
+                self.sbp_medication_column,
+                self.ldlc_medication_column,
+                self.sbp_medication_adherence_type_column,
+                self.ldlc_medication_adherence_type_column,
+            ]
+        ).get(event.index, query='alive == "alive"')
+
+        visitors = pop[
+            pop[self.visit_type_column].isin(
+                [
+                    data_values.VISIT_TYPE.EMERGENCY,
+                    data_values.VISIT_TYPE.SCHEDULED,
+                    data_values.VISIT_TYPE.BACKGROUND,
+                ]
+            )
+        ].index
+
+        pop.loc[visitors] = self.apply_sbp_treatment_ramp(pop_visitors=pop.loc[visitors])
+        pop.loc[visitors] = self.apply_ldlc_treatment_ramp(pop_visitors=pop.loc[visitors])
+
+        self.population_view.update(
+            pop[
+                [
+                    self.sbp_medication_column,
+                    self.ldlc_medication_column,
+                ]
+            ]
+        )
+
+    def treat_not_currently_medicated_sbp(
```

Comment:
There is a lot of code duplication between currently medicated and newly medicated. I think splitting them into separate functions at this level is inappropriate. If we want to create new functions for different paths (which I'm not completely convinced is necessary), I would do it only once we get to the treatment assignment nodes (i.e. the ones labeled B, C, and D). If you make rung 0 on the ladder `NO_TREATMENT`, nodes B and D look identical to me.